### PR TITLE
Update various dependences

### PR DESCRIPTION
**Pull request by saste**
Bunch of updates, locally tested.

### CODE REVIEW

**Comment by danyeaw** [comment]
@saste Thanks for the contribution. Can you please fixup the lint errors?

File: gvsbuild/projects/ffmpeg.py [hunk 1]
```diff
@@ -55,18 +55,11 @@ def build(self):
...
-        )
-        self.builder.exec_msys(
-            ["mv", "avutil.lib", "../lib/"],
-            working_dir=os.path.join(self.builder.gtk_dir, "bin"),
-        )
-        self.builder.exec_msys(
-            ["mv", "swscale.lib", "../lib/"],
-            working_dir=os.path.join(self.builder.gtk_dir, "bin"),
-        )
+        for lib in ["avcodec.lib", "avutil.lib", "swscale.lib"]:
```
Comment (danyeaw) [inline gvsbuild/projects/ffmpeg.py:58]: Nice simplification 👍

File: gvsbuild/projects/pkgconf.py [hunk 2]
```diff
@@ -27,10 +27,9 @@ def __init__(self):
             self,
             "pkg-config",
             prj_dir="pkgconf",
-            archive_url="https://distfiles.dereferenced.org/pkgconf/pkgconf-1.8.0.tar.gz",
-            hash="d7b6fdb522d81c11f5a0e0a0629a9f5480809ec90e595058674c1517822dfb8c",
+            archive_url="https://distfiles.dereferenced.org/pkgconf/pkgconf-1.9.3.tar.gz",
```
Comment (danyeaw) [inline gvsbuild/projects/pkgconf.py:30]: I'm going to revert this change, pkgconf 1.9.x is a testing version in preparation for version 2.0, we should still with 1.8.x until it is finalized.

**Comment by danyeaw** [comment]
@saste, thanks for the contribution! It looks like we also need to add a contributor guide for instructions to run `pre-commit install` so that lint errors will be fixed automatically when you go to commit.

**Comment by saste** [comment]
>  It looks like we also need to add a contributor guide for instructions to run pre-commit install so that lint errors will be fixed automatically when you go to commit.

At the same time, I find most of the black checks annoying (in fact making the code worse to read and to edit), I'd drop that kind of checks entirely and save development time and effort.

**Comment by nacho** [comment]
dunno, it is very simple to go over the lint and to fix them, I am happy that we have a lint and that we keep consistency. Thanks @danyeaw for that!
